src/main.c

```diff
@@ -502,7 +502,7 @@ u16 sub_08001D94(u16 *a)
     return a[0];
 }
 
-void sub_08001D98(u32 *a, struct Unk_08001D98 *b)
+void sub_08001D98(u32 *a, struct Unk_08001D98 *b) // TODO: Fix a's type. It's struct Unk_08001DE8_1 * actually.
```

Comment:
Are you sure? The way of accessing `a` in this function is contradicting the current declaration of `struct Unk_08001DE8_1`.

Reply:
I get this from the pseudo of sub_800B730.

  v8 = sub_8001D78(v7, (unsigned __int8)v2[15]);
  sub_800A630(v2[14], v2[13], (u32 *)v8);

sub_800A630:

sub_8001D98(a3, (Unk_08001D98 *)&v6);